contracts/test/v0.8/foundry/vrf/VRFV2Plus.t.sol

```diff
@@ -0,0 +1,364 @@
+
+    BlockhashStore s_bhs;
+    ExposedVRFCoordinatorV2Plus s_testCoordinator;
+    VRFV2PlusConsumerExample s_testConsumer;
+    MockLinkToken s_linkToken;
+    MockV3Aggregator s_linkEthFeed;
+
+    VRFCoordinatorV2Plus.FeeConfig basicFeeConfig =
+        VRFCoordinatorV2Plus.FeeConfig({fulfillmentFlatFeeLinkPPM: 0, fulfillmentFlatFeeEthPPM: 0});
+
+    // VRF KeyV2 generated from a node; not sensitive information.
+    // The secret key used to generate this key is: 10.
+    bytes vrfUncompressedPublicKey =
+        hex"a0434d9e47f3c86235477c7b1ae6ae5d3442d49b1943c2b752a68e2a47e247c7893aba425419bc27a3b6c7e693a24c696f794c2ed877a1593cbee53b037368d7";
+    bytes vrfCompressedPublicKey = hex"a0434d9e47f3c86235477c7b1ae6ae5d3442d49b1943c2b752a68e2a47e247c701";
+    bytes32 vrfKeyHash = hex"9f2353bde94264dbc3d554a94cceba2d7d2b4fdce4304d3e09a1fea9fbeb1528";
+
+    function setUp() public override {
+        BaseTest.setUp();
+
+        // Fund our users.
+        vm.roll(1);
+        vm.deal(LINK_WHALE, 10_000 ether);
+        changePrank(LINK_WHALE);
+
+        // Instantiate BHS.
+        s_bhs = new BlockhashStore();
+
+        // Deploy coordinator and consumer.
+        s_testCoordinator = new ExposedVRFCoordinatorV2Plus(address(s_bhs));
+
+        // Use create2 to deploy our consumer, so that its address is always the same
+        // and surrounding changes do not alter our generated proofs.
+        bytes memory consumerInitCode = bytes.concat(initializeCode, abi.encode(address(s_testCoordinator), LINK_WHALE));
+        bytes32 abiEncodedOwnerAddress = bytes32(uint256(uint160(LINK_WHALE)) << 96);
+        address consumerCreate2Address;
+        assembly {
+            consumerCreate2Address :=
+                create2(
+                    0, // value - left at zero here
+                    add(0x20, consumerInitCode), // initialization bytecode (excluding first memory slot which contains its length)
+                    mload(consumerInitCode), // length of initialization bytecode
+                    abiEncodedOwnerAddress // user-defined nonce to ensure unique SCA addresses
+                )
+        }
```

Comment:
I see, we want to always have it at the same address. To avoid this initcode stuff, can we change prank to a fixed address that will only be used to deploy this consumer, and do `s_testConsumer = new VRFV2PlusConsumerExample(address(s_testCoordinator), LINK_WHALE)` instead? Or would that not work? This approach right here works but seems cryptic, wondering if there's a simpler way.

Reply:
I was able to get rid of the big initcode hex actually, and now it just uses the type(Consumer).initCode. The address of a contract is based on the deployer's nonce when using the standard "deploy" opcode. So if we enforce that the consumer gets deployed right when we change prank it should be the same every time, but this is just a bit more flexible for future changes to the tests.